service/watcher/chartstatus/dynclient.go

```diff
@@ -27,19 +26,17 @@ func (c *ChartStatusWatcher) waitForDynClient(ctx context.Context) (dynamic.Inte
 		return c.k8sClient.DynClient(), nil
 	}
 
-	var chartOperatorAppCR *v1alpha1.App
+	var kubeConfigSecret *corev1.Secret
 	{
-		chartOperatorAppCR, err = c.waitForChartOperator(ctx)
+		kubeConfigSecret, err = c.waitForKubeConfig(ctx)
```

Comment:
I renamed the method because it's the kubeconfig we need. The chart-operator app CR is just how we get it.